python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -215,6 +220,109 @@ def categorical_embedding_with_vocabulary_file(feature_tensor, feature_info, fil
     return embedding
 
 
+class CategoricalDropout(layers.Layer):
+    """Custom Dropout class for categorical indices"""
+
+    def __init__(self, dropout_rate, seed=None, **kwargs):
+        """
+        Args:
+            dropout_rate: fraction of units to drop, i.e., set to OOV token 0
+            seed: random seed for sampling to mask/drop categorical labels
+
```

Comment:
@lastmansleeping Super minor question, but why are you using zeros instead of o's for "00V" references? Is this an industry standard (and if so, curious what the reasoning is)?

Reply:
EDIT Ah never mind, I see that it is o's, was just the font throwing me off! :)

Reply:
We need a numeric representation of OOV index. So any text that is not in our vocabulary will get mapped to 0th index.